### DISCUSSION TITLE
An incorrect type generation

### DISCUSSION THREAD

**Discussion by JustFly1984** [dotansimha/graphql-code-generator#6741] [Q&A]
in schema.graphql I have next types:

```graphql
"""A Page Item"""
interface PageItem {
  id: Int!
  rank: Float!
  type: String!
}

type ContentItem implements PageItem {
  id: Int!
  rank: Float!
  type: String!
 
  content: String!
}

type PhotoItem implements PageItem {
  id: Int!
  rank: Float!
  type: String!

  url: String!
  height: Int!
  width: Int!
}
```

then I run `graphql-codegen --config codegen.yml`

I get incorrect types generated next:

```typescript
  /** A Page Item */
export type PageItem = {
  id: Scalars['Int'];
  rank: Scalars['Float'];
  type: Scalars['String'];
};

export type ContentItem = PageItem & {
  id: Scalars['Int'];
  rank: Scalars['Float'];
  type: Scalars['String'];
  
  content: Scalars['String'];
};

export type PhotoItem = PageItem & {
  id: Scalars['Int'];
  rank: Scalars['Float'];
  type: Scalars['String'];

  url: Scalars['String'];
  height: Scalars['Int'];
  width: Scalars['Int'];
};
```

 I had to edit generated types to the next form to satisfy typescript logic:

```typescript
/** A Page Item */
export interface ContentItem {
  id: Scalars['Int'];
  rank: Scalars['Float'];
  type: Scalars['String'];

  content: Scalars['String'];
}

export interface PhotoItem {
  id: Scalars['Int'];
  rank: Scalars['Float'];
  type: Scalars['String'];

  url: Scalars['String'];
  height: Scalars['Int'];
  width: Scalars['Int'];
};

export type PageItem =
  | ContentItem
  | PhotoItem

```

Is it possible to get this behavior from type generator somehow?

My use-case is next.js react.js apollo client app.

Thank you for the great tool!

**Comment by ardatan** [comment]
Could you try `useImplementingTypes` flag?
https://www.graphql-code-generator.com/docs/plugins/typescript

**Reply by JustFly1984** [reply]
@ardatan useImplementingTypes: true produces same types as without it.

**Reply by JustFly1984** [reply]
@ardatan do you have other ideas?

**Reply by Urigo** [reply]
@JustFly1984 do you think you could create failing tests that show the issues?

**Reply by JustFly1984** [reply]
I'm not sure how to approach tests. The main issue here is that transformation of graphql `implements` should not be resulted in typescript `&`, but the set of types instead, cos logic in both languages not overlaps.

**Reply by JustFly1984** [reply]
@Urigo @ardatan 
We are using this generator in multiple projects, and currently I'm copying generated ts file, and editing the copy to correct this behavior and also to replace inlined types.
Generated files are huge - almost 10k lines. If I replace types correctly, it improves VSCode performance, and DX, cos currently the project is lagging at mouse hover types inference.
Is it possible to force generator to generate types without repeated types inlined all other the file?

**Reply by Urigo** [reply]
@JustFly1984 maybe the first step for you is to fork codegen and make the change you need.
That way you will have the changes you need in your codebase.
Later you can submit a draft PR of your changes so we could discuss if and how to include them on upstream
